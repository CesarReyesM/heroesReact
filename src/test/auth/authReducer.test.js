import { shallow } from 'enzyme';
import { authReducer } from '../../auth/authReducer';
import '../../setupTest';
import { types } from '../../types/types';

describe('Probando authReducer', () => {

    let wrapper = shallow(<authReducer/>)

    test('debe de conectar', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de retornar el estado por defecto', () => {
        const state = authReducer({logged: false}, {})

        expect(state).toEqual({logged:false})
    })

    test('debe de autenticar y colocar el name del usuario', () => {

        const action = {
                type: types.login,
                payload:{
                    name:'Cesar'
                }
        }
        const state = authReducer({logged: false}, action);
    
        expect(state).toEqual({logged:true, name:'Cesar'})

    });

    test('Debe de borrar el name del usuario yy logged en false', () => {

        const action = {
            type: types.logout
        }

        const state = authReducer({logged: true, name:'Cesar'}, action);

        expect(state).toEqual({logged:false})

    })




})