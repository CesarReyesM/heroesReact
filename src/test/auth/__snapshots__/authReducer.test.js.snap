// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando authReducer debe de conectar 1`] = `<authReducer />`;

exports[`Probando authReducer debe de retornar el estado por defecto 1`] = `<authReducer />`;
